tests/test-rest-posts-controller.php

```diff
@@ -1664,6 +1665,50 @@ public function test_get_additional_field_registration() {
 		$wp_rest_additional_fields = array();
 	}
 
+	public function test_enforce_minimum_and_max_to_posts_per_page() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array(
+			'per_page' => 101,
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array(
+			'per_page' => -1,
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array(
+			'filter' => array(
+				'posts_per_page' => 101
+			)
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array(
+			'filter' => array(
+				'posts_per_page' => -1
+			)
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array(
+			'per_page' => 10,
+			'filter' => array(
+				'posts_per_page' => -1
+			)
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+	}
+
```

Comment:
I would also verify in the tests that the error response message matches and that you are not getting another 400 error.  Otherwise looks good!  So `$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );` after any of your status checks for an error.